frontend/src/app/main/data/workspace/text/shortcuts.cljs

```diff
@@ -0,0 +1,218 @@
+        bold-options              (cond
+                                    (str/includes? current-variant "black")
+                                    ["black" "bold" "700"]
+                                    (str/includes? current-variant "700")
+                                    ["700" "black" "bold"]
+                                    :else
+                                    ["bold" "black" "700"])
+        current-variant-no-italic (cond
+                                    (str/includes? current-variant "italic")
+                                    (subs current-variant 0 (- (count current-variant) 6))
+                                    (str/includes? current-variant "cursive")
+                                    (subs current-variant 0 (- (count current-variant) 7))
+                                    :else nil)
+        regular-options           [current-variant-no-italic "regular" "normal" "400"]
+        italic-options            [(when (and (not (str/includes? current-variant "bold"))
+                                              (not (str/includes? current-variant "black"))
+                                              (not (str/includes? current-variant "700")))
+                                     (str current-variant "italic"))
+                                   "italic" "cursive"]
+        bold-italic-options       (cond
+                                    (str/includes? current-variant "black")
+                                    ["blackitalic" "blackcursive" "bolditalic" "700italic" "boldcursive" "700cursive"]
+                                    (str/includes? current-variant "700")
+                                    ["700italic" "700cursive" "bolditalic"  "blackitalic" "boldcursive"  "blackcursive"]
+                                    :else
+                                    ["bolditalic" "700italic" "blackitalic" "boldcursive" "700cursive" "blackcursive"])
+        is-bold?                  (fn [variant-id] (some #(str/includes? variant-id %) bold-options))
+        is-italic?                (fn [variant-id] (some #(str/includes? variant-id %) italic-options))
+        font-id                   (:font-id text-values)
+        fonts                     (deref fonts/fontsdb)
+        font                      (get fonts font-id)
+        variants                  (map :id (:variants font))
+        choose-regular            (fn [] (first-intersection variants regular-options))
+        choose-bold               (fn [] (first-intersection variants bold-options))
+        choose-italic             (fn [] (first-intersection variants italic-options))
+        choose-bold-italic        (fn [] (or (first-intersection variants bold-italic-options) (choose-bold)))
+        choose-italic-bold        (fn [] (or (first-intersection variants bold-italic-options) (choose-italic)))
+
+        new-variant               (if (= variant-id "bold") ;; it can only be bold or italic
```

Comment:
I don't like all these nested ifs. There is 4 options regular, italic, bold, bold+italic so I think there should be just 1 `cond` with 4 leaves... my proposal (please, check to make sure I didn't mess up)

```clojure
(let [set-bold?   (= variant-id "bold")
      set-italic? (= variant-id "italic")
      bold?    (is-bold? current-variant)
      italic?  (is-italic? current-variant)]
  (cond
    ;; bold
    (or (and set-bold? (not bold?) (not italic?))
        (and set-italic? bold? italic?))
    (choose-bold)

    ;; italic
    (or (and set-bold? bold? italic?)
        (and set-italic? (not bold?) (not italic?)))
    (choose-italic)
    
    ;; bold+italic
    (or (and set-bold? (not bold?) italic?)
        (and set-italic? (not italic?) bold?))
    (choose-italic-bold)
    
    ;; regular
    :else
    (choose-regular)))
   
 ```

Reply:
I've done it a little different, but I think it's now clear :)